lib/inputstreamhelper/__init__.py

```diff
@@ -389,8 +389,17 @@ def _http_get(self, url):
         # log('Response: {response}', response=content)
         return content.decode()
 
-    def _http_download(self, url, message=None):
+    def _http_download(self, url, message=None, checksum=None, hash_alg='sha1'):
         """Makes HTTP request and displays a progress dialog on download."""
+        if checksum:
+            from hashlib import sha1, md5
+            if hash_alg == 'sha1':
+                calc_checksum = sha1()
+            elif hash_alg == 'md5':
+                calc_checksum = md5()
+            else:
+                raise ValueError('Invalid hash algorithm specified: {}'.format(hash_alg))
```

Comment:
Not sure if we want this to blow up like this. Since it now works without checksum verification, maybe we ought to just report it, but still continue without verification.

Reply:
Ok, changed it.